ipwb/indexer.py

```diff
@@ -123,6 +124,18 @@ def index_file_at(warc_paths, encryption_key=None,
     for warc_path in warc_paths:
         verify_file_exists(warc_path)
 
+    warc_paths_to_append = []
+    warc_paths_to_remove = []
+    for warc_path in warc_paths:
+        if is_wacz(warc_path):
+            warc_paths_to_append += extract_warcs_from_wacz(warc_path)
+            warc_paths_to_remove.append(warc_path)
+
+    # Manipulate list of WARCs extracted from WACZ
+    for ptr in warc_paths_to_remove:
+        warc_paths.remove(ptr)
+    warc_paths = warc_paths + warc_paths_to_append
```

Comment:
I have noticed that in some WACZ files sometimes there are `.warc` files, which are actually `.warc.gz` files. I hope `warcio` transparently handles those.

Reply:
This would make a good test case.

Reply:
I think we do not need to define `warcs_to_cleanup_post_indexing` and use the existing `warc_paths_to_append` in its place.

Reply:
I think we should rename `warc_paths_to_remove` to something like `wacz_paths`, because the use of the term `remove` can now be confusing if it means the path will be removed from being processed or the file will be removed (especially, after introducing the cleanup function that removes files).

Reply:
Renaming and conversion to temp directories introduced in [e4b9998](https://github.com/oduwsdl/ipwb/pull/770/commits/e4b9998f82f7cb75f63ed808ec5080c0e7449074).